spring/boot3-autoconfigure/src/main/java/com/linecorp/armeria/spring/InternalServices.java

```diff
@@ -122,12 +122,14 @@ public static InternalServices of(
 
             final String legacyPrometheusMeterRegistryClassName =
                     "io.micrometer.prometheus.PrometheusMeterRegistry";
-            final boolean hasLegacyPrometheus = hasAllClasses(
-                    legacyPrometheusMeterRegistryClassName,
-                    "io.prometheus.client.CollectorRegistry");
+            if (expositionService == null) {
```

Comment:
It is likely a bug where `expositionService` created by `PrometheusSupport` is overwritten by `PrometheusLegacySupport`. So I changed to use `PrometheusLegacySupport` if `expositionService` is created by `PrometheusSupport`.